submission_runner.py

```diff
@@ -453,6 +453,8 @@ def train_once(
                   save_intermediate_checkpoints=FLAGS
                   .save_intermediate_checkpoints)
 
+          if USE_PYTORCH_DDP:
```

Comment:
Should this really only be called when PyTorch DDP is used or also when just DataParallel or a single GPU is used?

Reply:
hmmm that's a good question, if it's OOMing in the other cases (DataParallel or single GPU) we should use the condition you suggest 

right now I actually don't know if it OOMs in those other cases, maybe we just go ahead and do it for all cases to be safe ?

Reply:
okay I'm gonna merge this for now and test OOM for other cases, will send a fix

Reply:
Yeah let's just do it for all cases, I don't see any downside to it.